activerecord/test/cases/strict_loading_test.rb

```diff
@@ -63,6 +59,18 @@ def test_strict_loading_n_plus_one_only_mode
     assert_raises ActiveRecord::StrictLoadingViolationError do
       developer.projects.last.firm
     end
+  end
+
+  def test_strict_loading_n_plus_one_only_mode_with_belongs_to
+    developer = Developer.first
+    ship = Ship.first
+    ShipPart.create!(name: "Stern", ship: ship)
+
+    ship.update_column(:developer_id, developer.id)
+    developer.reload
+
+    developer.strict_loading!(mode: :n_plus_one_only)
+    assert_predicate developer, :strict_loading?
 
     # Does not raise when a belongs_to association (:ship) loads its
```

Comment:
Hmm, I'm not sure this is correct.
The Developer has one Ship, it's no a `belongs_to`.

Reply:
I think developer belongs_to firm here

Reply:
I think the point is that it's a singular association. `:n_plus_one_only` mode works the same for `has_one` and `belongs_to`: you can keep calling lazy loading those because they won't result in a N+1.